src/batou/secrets/__init__.py

```diff
@@ -1,77 +1,606 @@
+        if recipients.value is None:
+            return []
+        recipients = re.split(r"(\n|,)+", recipients.value)
+        recipients = [r.strip() for r in recipients if r.strip()]
+        return recipients
+
+    def write_config(self, content: bytes):
+        config = ConfigUpdater().read_string(content.decode("utf-8"))
+        secret_provider = config.get("batou", "secret_provider", fallback=None)
+        if secret_provider is None or secret_provider.value is None:
+            config.set("batou", "secret_provider", "gpg")
+            print("Setting secret provider to gpg.")
+        secret_provider = config.get("batou", "secret_provider")
+        if secret_provider.value != "gpg":
+            self.change_secret_provider(config, self)
+            return
+        recipients_opt = config.get("batou", "members")
+        if recipients_opt is None or recipients_opt.value is None:
+            raise ValueError(
+                "Please add a 'batou.members' section to the secrets file."
+            )
+        recipients = re.split(r"(\n|,)+", recipients_opt.value)
+        recipients = [r.strip() for r in recipients if r.strip()]
+        if not recipients or len(recipients) == 0 or recipients[0] == "":
+            raise ValueError(
+                "Please add at least one recipient to the secrets file."
+            )
+        self.config_file.write(
+            str(config).encode("utf-8"),
+            recipients,
+        )
+        self.write_secret_files(self.read_secret_files())
+
+    def secret_provider_str(self) -> str:
+        return "gpg"
+
+
+def process_age_recipients(members, environment_path):
+    """Process the recipients list."""
+    key_meta_file_path = os.path.join(
+        environment_path,
+        "age_keys.txt",
+    )
+    key_meta_file_content = """\
+###########################################
+# This file is automatically generated by #
+# batou. It contains the public keys of   #
+# the members of the environment. It is   #
+# re-written every time a secrets file is #
+# encrypted (after a change).             #
+###########################################
+"""
+    new_members = []
+    for key in members:
+        key = key.strip()
+        if key.startswith("ssh-"):
+            # it's a plain ssh public key and we can simply add it
+            new_members.append(key)
+            key_meta_file_content += f"# plain ssh public key\n{key}\n"
+        elif key.startswith("age1"):
+            # it's a plain age public key and we can simply add it
+            new_members.append(key)
+            key_meta_file_content += f"# plain age public key\n{key}\n"
+        elif key.startswith("https://") or key.startswith("http://"):
+            # it's a url to a key file, so we need to download it
+            # and add it to the key meta file
+            if key.startswith("http://"):
+                print("WARNING: Downloading public keys over http is insecure!")
```

Comment:
I'm not fully convinced even allowing HTTP is a good choice, but I'm happy to sit this one out.